common/core/src/main/java/com/vmware/connectors/common/payloads/response/Card.java

```diff
@@ -353,16 +367,68 @@ public Builder addTag(String tag) {
             return this;
         }
 
+        /**
+         * Add SHA-1 of card fields.
+         *
+         * @param hash SHA-1 of card header, body, and action fields.
+         * @return this Builder instance, for method chaining
+         */
+        public Builder addHash(String hash) {
+            card.hash = hash;
+            return this;
+        }
+
         /**
          * Return the Card under construction and reset the Builder to its initial state.
          *
          * @return The completed Card
          */
         @SuppressWarnings("PMD.UnnecessaryLocalBeforeReturn")
         public Card build() {
+            // If the connector author has already set the hash, then do not override the hash value.
+            if (StringUtils.isBlank(card.hash)) {
+                card.hash = computeHash();
+            }
+
             Card completedCard = this.card;
             reset();
             return completedCard;
         }
+
+        public String computeHash() {
+            final StringBuilder result = new StringBuilder();
+            if (StringUtils.isNotBlank(card.name)) {
+                result.append(card.name);
+            }
+
+            if (card.template != null && StringUtils.isNotBlank(card.template.getHref())) {
+                result.append(card.template.getHref());
+            }
+
+            if (card.header != null) {
+                result.append(card.header.hash());
+            }
+
+            if (card.body != null) {
+                result.append(card.body.hash());
+            }
+
+            if (!CollectionUtils.isEmpty(card.actions)) {
+                card.actions.stream()
+                        .filter(Objects::nonNull)
+                        .forEach(cardAction -> result.append(cardAction.hash()));
+            }
+
+            if (card.image != null && StringUtils.isNotBlank(card.image.getHref())) {
+                result.append(card.image.getHref());
+            }
+
+            if (!CollectionUtils.isEmpty(card.tags)) {
+                final Set<String> sortedSet = new TreeSet<>(card.tags);
```

Comment:
Instead of using TreeSet because it happens to be sorted, just use java 8:

```java
card.tags.stream()
        .sorted()
        .forEach(tag -> result.append(tag));
```

Reply:
Done.